src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -210,7 +329,73 @@ private void OnDiagnosticsUpdated(object sender, DiagnosticsUpdatedArgs e)
             private void OnDiagnosticsUpdatedOnBackground(DiagnosticsUpdatedArgs e)
             {
                 this.AssertIsBackground();
-                RegisterNotification(() => OnDiagnosticsUpdatedOnForeground(e));
+                if (_disposed)
+                {
+                    return;
+                }
+
+                var document = e.Solution.GetDocument(e.DocumentId);
+                if (document == null)
+                {
+                    // Not a diagnostic event for a document.  Not something we can handle.
```

Comment:
❓ Do you never get a "clear all" event which contains a project ID but no document ID, and used to clear all diagnostics?

Reply:
I truly hope not.  @heejaechang ?

Reply:
➡️ It looks like it can be null, but it's used to clear diagnostics not associated with any particular document. You still get per-document events for all of the diagnostics the tagger(s) will care about.